### PR TITLE
Update to python 3.7

### LINKED ISSUE #56873: pycurl dependency on Windows unable to be satisfied
**Description**
Doing a pip install of salt on Windows (10 ver 1903) with python 3.7 fails due to not being able to meet the pycurl requirement. 

```
pip install salt
...
Collecting pycurl==7.43.0.2
  Using cached pycurl-7.43.0.2.tar.gz (214 kB)
    ERROR: Command errored out with exit status 10:
     command: 'c:\python37\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\mchug\\AppData\\Local\\Temp\\pip-install-p1cutikh\\pycurl\\setup.py'"'"'; __file__='"'"'C:\\Users\\mchug\\AppData\\Local\\Temp\\pip-install-p1cutikh\\pycurl\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\mchug\AppData\Local\Temp\pip-install-p1cutikh\pycurl\pip-egg-info'
         cwd: C:\Users\mchug\AppData\Local\Temp\pip-install-p1cutikh\pycurl\
    Complete output (1 lines):
    Please specify --curl-dir=/path/to/built/libcurl
    ----------------------------------------
ERROR: Command errored out with exit status 10: python setup.py egg_info Check the logs for full command output.
```

**Setup**
Required pycurl 7.43.0.2 is not installable. However, newer pycurl 7.43.1 does work.

**Steps to Reproduce the behavior**
Run `pip install salt` on windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Updates Windows installer scripts to use Python 3.7.4 as the version of python to be bundled for Windows. Removes references to Python 2. Updates a few dependencies required by Python 3.7.

Merge this one first: https://github.com/saltstack/salt/pull/56120

Fixes: https://github.com/saltstack/salt/issues/56873

### Previous Behavior
Python 2.7.15 and Python 3.5.4

### New Behavior
Python 3.7.4

### Tests written?
No

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by s0undt3ch — DISMISSED** [review]

**Review by dwoz — CHANGES_REQUESTED** [review]
Please separate this into two PRs. One for the python3 change and one for the python2 change.

**Comment by markuskramerIgitt** [comment]
### Regarding salt/version.py 
Hi @twangboy  please replace commit 

    Update msi_conformant_version for 3000  6837d07

by the commits of https://github.com/saltstack/salt/pull/56352/commits

- The function `msi_conformant_version()` should be removed
- This should resolve the version.py conflicts

**Review by marbx — CHANGES_REQUESTED** [review]
Same comment as in https://github.com/saltstack/salt/pull/56120

### Regarding salt/version.py 
Hi @twangboy  please replace commit 

    Update msi_conformant_version for 3000  6837d07

by the commits of https://github.com/saltstack/salt/pull/56352/commits

- The function `msi_conformant_version()` should be removed
- This should resolve the version.py conflicts

File: pkg/windows/req.txt [hunk 1]
```diff
@@ -6,9 +6,7 @@ cffi==1.12.2
 CherryPy==17.4.1
 cryptography==2.6.1
 distro==1.4.0
-enum34==1.1.6
```
Comment (markuskramerIgitt) [inline pkg/windows/req.txt:8]: Nice cleanup

**Review by Ch3LL — DISMISSED** [review]

**Review by dwoz — DISMISSED** [review]

**Comment by dwoz** [comment]
@twangboy we've got conflicts again.

**Comment by twangboy** [comment]
@dwoz Fixed conflicts

**Pull request merged**